app-header/app-header.html

```diff
@@ -0,0 +1,218 @@
+
+Example:
+
+Add a drawer in the left side of the screen:
+
+    <app-header condenses>
+      <app-toolbar>
+        <div title>App name</div>
+      </app-toolbar>
+    </app-drawer>
+
+@demo demo.html
+-->
+
+<dom-module id="app-header">
+
+  <style>
+
+    :host {
+      display: block;
+    }
+
+    :host::after {
+      content: '';
+      position: absolute;
+      left: 0px;
+      right: 0px;
+      bottom: -5px;
+      width: 100%;
+      height: 5px;
+      pointer-events: none;
+      opacity: 0;
+      box-shadow: inset 0px 5px 6px -3px rgba(0, 0, 0, 0.4);
+      -webkit-transition: opacity 0.2s;
+      transition: opacity 0.2s;
+      /* fix bug in chrome */
+      border-top: 1px solid transparent;
+    }
+
+    :host([shadow])::after {
+      opacity: 1; 
+    }
+
+  </style>
+
+  <template>
+
+    <content></content>
+
+  </template>
+
+  <script>
+
+    Polymer({
+
+      is: 'app-header',
+
+      properties: {
+
+        /**
+         * If true, the header's height will condense as the user scrolls down
+         * from the top of the content area.
+         */
+        condenses: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, the header is fixed to the top and never moves away.
+         */
+        fixed: {
+          type: Boolean,
+          value: false,
+          observer: '_fixedChanged'
+        },
+
+        /**
+         * If true, the condensed header is always shown and does not move away.
+         */
+        alwaysVisible: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, a shadow will appear when the user is not at the top of the content area.
+         */
+        waterfall: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * If true, a shadow will appear when the user is not at the top of the content area.
+         */
+        shadow: {
+          type: Boolean,
+          readOnly: true,
+          reflectToAttribute: true,
+          value: false
+        }
+
+      },
+
+      behaviors: [
+        Polymer.CustomScrollerBehavior
+      ],
+
+      /**
+       * A cached offsetHeight of the current element
+       */
+      _height: 0,
+
+      /**
+       * The top most app-toolbar
+       */
+      _topbar: null,
+
+      /**
+       * The distance in pixels the header will be translated to when scrolling.
+       */
+      _deltaHeight: 0,
+
+      get _clampedScrollTop() {
+        return Math.max(0,
+            this.alwaysVisible ? Math.min(this._scrollTop, this._deltaHeight) : this._scrollTop);
+      },
+
+      get _headerHeight() {
+        return this.alwaysVisible ? this._deltaHeight : this._height;
+      },
+
+      created: function() {
+        this._oldY = 0;
+        this._oldTop = 0;
+      },
+
+      attached: function() {
+        // set default scroller to window
```

Comment:
the comment seems not applied anymore, maybe update the comment or just remove it